driver-core/src/test/java/com/datastax/driver/core/DataTypeTest.java

```diff
@@ -43,19 +42,66 @@ static boolean exclude(DataType t) {
         return t.getName() == DataType.Name.COUNTER;
     }
 
-    private static String[] getCQLStringTestData(DataType dt) {
+    private static String[] getOutputCQLStringTestData(DataType dt) {
         switch (dt.getName()) {
             case ASCII:
             case TEXT:
             case VARCHAR:
                 return new String[]{ "'foo'", "'fo''o'" };
             case BIGINT:
+                return new String[]{ "42", "91294377723", "-133" };
             case TIMESTAMP:
+                return new String[]{ "42", "91294377723", "-133", "784041330999" };
+            case DATE:
+                return new String[]{ "'2014-01-01'", "'1970-01-01'", "'1970-01-01'", "'-5877641-06-23'" };
+            case TIME:
+                return new String[]{ "'15:00:12.123450000'", "'00:00:00.000000000'", "'15:00:12.123450000'" };
+            case BLOB:
+                return new String[]{ "0x2450", "0x" };
+            case BOOLEAN:
+                return new String[]{ "true", "false" };
+            case DECIMAL:
+                return new String[]{ "1.23E+8" };
+            case DOUBLE:
+                return new String[]{ "2.39324324", "-12.0" };
+            case FLOAT:
+                return new String[]{ "2.39", "-12.0" };
+            case INET:
+                return new String[]{ "'128.2.12.3'" };
+            case TINYINT:
+                return new String[]{ "-4", "44" };
+            case SMALLINT:
+                return new String[]{ "-3", "43" };
+            case INT:
+                return new String[]{ "-2", "42" };
+            case TIMEUUID:
+                return new String[]{ "fe2b4360-28c6-11e2-81c1-0800200c9a66" };
+            case UUID:
+                return new String[]{ "fe2b4360-28c6-11e2-81c1-0800200c9a66", "067e6162-3b6f-4ae2-a171-2470b63dff00" };
+            case VARINT:
+                return new String[]{ "12387290982347987032483422342432" };
+            default:
+                throw new RuntimeException("Missing handling of " + dt);
+        }
+    }
+
+    private static String[] getInputCQLStringTestData(DataType dt) {
+        switch (dt.getName()) {
+            case ASCII:
+            case TEXT:
+            case VARCHAR:
+                return new String[]{ "'foo'", "'fo''o'" };
+            case BIGINT:
                 return new String[]{ "42", "91294377723", "-133" };
+            case TIMESTAMP:
+                // single quotes are optional for long literals, mandatory for date patterns
+                return new String[]{ "42", "91294377723", "-133", "'1994-11-05T14:15:30.999+0100'" };
```

Comment:
Note that `1994-11-05T14:15:30.999+0100` is not fully ISO-8601-compliant, but that's what the driver currently accepts.